### PR TITLE
hwatch: init at 0.3.6

### PR DESCRIPTION
**Pull request by hamburger1984**
###### Description of changes

This adds a new tool [`hwatch`](https://github.com/blacknon/hwatch) to nixpkgs.

It is..

> A modern alternative to the watch command, records the differences in execution results and can check this differences at after. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/hwatch/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub, fetchpatch, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hwatch";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "blacknon";
+    repo = pname;
+    rev = "d86479f3b101be7c7b4ed0da3f008b5f2bf583fe";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/hwatch/default.nix:10]: Why are we not using the version revision?

File: pkgs/tools/misc/hwatch/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "sha256-dGntqLq+wDEsYU1HInX4g7De+WnDZcFpep/zxudlNSk=";
+  };
+
+  cargoPatches = [
```
Comment (legendofmiracles) [inline pkgs/tools/misc/hwatch/default.nix:14]: Can you add a comment for future maintainers why the patch is needed?

File: pkgs/tools/misc/hwatch/default.nix (hunk 1 above)
Comment (hamburger1984) [reply pkgs/tools/misc/hwatch/default.nix]: Adjusted to "rev = version"

File: pkgs/tools/misc/hwatch/default.nix (hunk 2 above)
Comment (hamburger1984) [reply pkgs/tools/misc/hwatch/default.nix:14]: Definitely

File: pkgs/tools/misc/hwatch/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  cargoSha256 = "sha256-cFpfdlItcbYtO6788OEKMm3nmEzfVFsdgrcF4mdlFwk=";
+
+  meta = with lib; {
+    homepage = "https://github.com/blackmon/hwatch";
+    description= "Modern alternative to the watch command";
+    longDescription = ''
+      A modern alternative to the watch command, records the differences in execution results and can check this differences at after
+    '';
+    license = licenses.mit;
+    platforms = platforms.linux;
```
Comment (fabaff) [inline pkgs/tools/misc/hwatch/default.nix:31]:
```suggestion
    maintainers = with maintainers; [ ];
    platforms = platforms.linux;
```

New packages need a maintainer, see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

File: pkgs/tools/misc/hwatch/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    })
+  ];
+  cargoSha256 = "sha256-cFpfdlItcbYtO6788OEKMm3nmEzfVFsdgrcF4mdlFwk=";
+
+  meta = with lib; {
+    homepage = "https://github.com/blackmon/hwatch";
+    description= "Modern alternative to the watch command";
+    longDescription = ''
+      A modern alternative to the watch command, records the differences in execution results and can check this differences at after
+    '';
```
Comment (fabaff) [inline pkgs/tools/misc/hwatch/default.nix:29]: Usually `longDescription` has a period at the end unlike `description`.

```suggestion
    longDescription = ''
      A modern alternative to the watch command, records the differences in
      execution results and can check this differences at after.
    '';
```

File: pkgs/tools/misc/hwatch/default.nix (hunk 3 above)
Comment (hamburger1984) [reply pkgs/tools/misc/hwatch/default.nix:33]: I'll be the maintainer then.

**Comment by hamburger1984** [comment]
Rebased & squashed commits into one for hwatch and one to add maintainer to list.

**Comment by hamburger1984** [comment]
> `0.3.6` was released 3 hours ago, and thx for the PR to adding Cargo.lock! https://github.com/blacknon/hwatch/releases/tag/0.3.6

Thanks for merging - will update this.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 177188` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hwatch</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see hhttps://github.com/NixOS/nixpkgs/pull/177188#issuecomment-1153227168)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...iew/pr-177188]$ ./results/hwatch/bin/hwatch 
error: The following required arguments were not provided:
    <command>...
[...]
```

Looks good to me.

**Pull request merged**